client/go/commands/node_add.go

```diff
@@ -0,0 +1,163 @@
+	"github.com/lpabon/godbc"
+	"net/http"
+	"os"
+	"time"
+)
+
+type NodeAddCommand struct {
+	Cmd
+	options            *Options
+	zone               int
+	managmentHostNames string
+	storageHostNames   string
+	clusterId          string
+}
+
+func NewNodeAddCommand(options *Options) *NodeAddCommand {
+
+	godbc.Require(options != nil)
+
+	cmd := &NodeAddCommand{}
+	cmd.name = "add"
+	cmd.options = options
+	cmd.flags = flag.NewFlagSet(cmd.name, flag.ExitOnError)
+	cmd.flags.IntVar(&cmd.zone, "zone", 0, "The zone in which the node should reside")
+	cmd.flags.StringVar(&cmd.clusterId, "cluster", "", "The cluster in which the node should reside")
+	cmd.flags.StringVar(&cmd.managmentHostNames, "managment-host-name", "", "Managment host name")
+	cmd.flags.StringVar(&cmd.storageHostNames, "storage-host-name", "", "Storage host name")
+
+	//usage on -help
+	cmd.flags.Usage = func() {
+		fmt.Println(`
+Node add is a command used for adding a node to the specified cluster.
+
+USAGE
+	heketi node add [options]
+
+OPTIONS`)
+
+		//print flags
+		cmd.flags.PrintDefaults()
+		fmt.Println(`
+EXAMPLES
+	Add a node to heketi
+		$ heketi -server http://localhost:8080 node add \
+	  		 -zone 3 \
+	  		 -cluster 3e098cb4407d7109806bb196d9e8f095 \
+	  		 -managment-host-name node1-manage.gluster.lab.com \
+	  		 -storage-host-name node1-storage.gluster.lab.com`)
+	}
+	godbc.Ensure(cmd.flags != nil)
+	godbc.Ensure(cmd.name == "add")
+
+	return cmd
+}
+
+func (a *NodeAddCommand) Name() string {
+	return a.name
+
+}
+
+func (a *NodeAddCommand) Exec(args []string) error {
+
+	//parse args
+	a.flags.Parse(args)
+
+	//ensure we have Url
+	if a.options.Url == "" {
+		fmt.Fprintf(stdout, "You need a server!\n")
+		os.Exit(1)
+	}
+
+	s := a.flags.Args()
+	if len(s) != 0 {
+		return errors.New("Too many arguments!")
+	}
+	//set url
+	url := a.options.Url
+
+	//create request blob
+	requestBlob := glusterfs.NodeAddRequest{}
+	requestBlob.ClusterId = a.clusterId
+	requestBlob.Hostnames.Manage = []string{a.managmentHostNames}
+	requestBlob.Hostnames.Storage = []string{a.storageHostNames}
+	requestBlob.Zone = a.zone
+
+	//marshal blob to get []byte to be Post'd
+	request, err := json.Marshal(requestBlob)
+	if err != nil {
+		return errors.New("json marshaling did not work")
+	}
+	//do Post and check if sent to server
+	r, err := http.Post(url+"/nodes", "application/json", bytes.NewBuffer(request))
+	if err != nil {
+		fmt.Fprintf(stdout, "Error: Unable to send command to server: %v", err)
+		return err
+	}
+
+	//check status code
+	if r.StatusCode != http.StatusAccepted {
+		utils.GetStringFromResponseCheck(r)
+	}
+
+	//Query queue until finished
+	location, err := r.Location()
+	for {
+		r, err := http.Get(location.String())
+		if err != nil {
+			return err
+		}
+		if r.Header.Get("X-Pending") == "true" {
+			if r.StatusCode == http.StatusOK {
+				time.Sleep(time.Millisecond * 3)
+				continue
+			} else {
+				utils.GetStringFromResponseCheck(r)
```

Comment:
This function only returns the string inside the Body.  We should then create an error from the string and return it.  This may need to be a utility function since the same sequence is used in many places.